src/Controller/Admin/Asset/AssetController.php

```diff
@@ -398,9 +398,21 @@ public function addAssetCompatibilityAction(Request $request, Config $config): J
     public function existsAction(Request $request): JsonResponse
     {
         $parentAsset = \Pimcore\Model\Asset::getById((int)$request->get('parentId'));
+        if ($dir = $request->get('dir')){
+            // this is for uploading folders with Drag&Drop
+            // param "dir" contains the relative path of the file
+            if (strpos($dir, '..') !== false) {
+                throw new \Exception('not allowed');
+            }
+            $dir =  '/' . trim($dir, '/ ');
+        } else {
+            $dir = '';
```

Comment:
I guess we could remove the else part and make the dir empty by default?

Reply:
`$dir` is set above from the request param so even if it was initialized to empty string it would be re-set to null. or did you mean something else?

Reply:
Ah sorry, seems I missed that it was not equal but assigning in the if statement.

I personally dislike these kind of if statements since they only work in PHP. 

I'd rather get the `$dir` first from the `$request` and then checking if something is in the request.
At least for me it would enhance the readability.

E.g.
```php
// Initialize "dir" with an empty string by default
$dir = '';

// Get the "dir" parameter from the request, if available
if ($request->get('dir')) {
    $dir = $request->get('dir');
    ...
}
```

Reply:
Why not use the default argument from the get function?
```
$dir = $request->query->get('dir', '');

// Get the "dir" parameter from the request, if available
if ($dir) {
    ...
}
```